### ISSUE TITLE
Regression in detecting unused parameters in 5.5.0-beta

### ISSUE THREAD

**Issue by danvk** [microsoft/TypeScript#58493]
### 🔎 Search Terms

- 5.5 predicate
- 6133

### 🕗 Version & Regression Information

- This changed between versions 5.4.5 and 5.5.0-beta

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.5.0-beta#code/GYVwdgxgLglg9mABABwE4FMAmB1GUAWAqmCAM5YAUAhgFyLgDWYcA7mAJSIDeAUIv4gxQQqJAEIxAWSoEAdKiphMcALYV2Abh4BfHkA

### 💻 Code

```ts
function predWithUnused(a: unknown) {
    return !!Math.random();
}
```

### 🙁 Actual behavior

TS 5.5.0-beta reports that `a` is used.

### 🙂 Expected behavior

It should be reported as unused.

### Additional information about the issue

Compare 5.4.5

<img width="751" alt="image" src="https://github.com/microsoft/TypeScript/assets/98301/1e7ddc0e-46a2-4494-ace5-126e953a6c6b">

vs. 5.5.0-beta:

<img width="509" alt="image" src="https://github.com/microsoft/TypeScript/assets/98301/caf6727f-1f9a-4b4c-855c-a0478e069010">

This was almost certainly broken by #57465. It adds a call to `getTypeOfSymbol` for every parameter of a candidate predicate function in `checkIfExpressionRefinesAnyParameter`. This marks each of those parameters as referenced, even though there's no reason to assume that they are, since this call is speculative.

This also affects the "Delete all unused declarations" quickfix.

**Comment by Pulkit1822** [comment]
@danvk , I would like to fix this it.
